### PR TITLE
remove the printing lag

### PR DESCRIPTION
**Pull request by brandondutra**
some people think “Analyzing file /content/datalab/so/cleaned_train.csv” is printed out too late. and that I'm missing a sys.stdout.flush(). But I was already doing this!

Looking into this, I noticed the lag does not exist in the terminal version of the code, it only happens in the magic. The solution required two changes

1. in DL's subprocess, it should flush after printing
1. in analyze, output needs to have newlines, because p.stdout.readline() waits for a newline. This means we cannot do the "doing x..done" pattern. I now do "doing x...\n x is done" pattern.

I'm also printing when we are expanding the file pattern for local run as this can take a while if the files are on gcs.

### CODE REVIEW

File: solutionbox/code_free_ml/mltoolbox/code_free_ml/analyze.py [hunk 1]
```diff
@@ -316,10 +316,14 @@ def run_local_analysis(output_dir, csv_file_pattern, schema, inverted_features):
   Raises:
     ValueError: on unknown transfrorms/schemas
   """
+  sys.stdout.write('Expanding any file patterns...\n')
```
Comment (qimingj) [inline solutionbox/code_free_ml/mltoolbox/code_free_ml/analyze.py]: Do we need this printed? It seems expanding files should be quick.

Comment (brandondutra) [inline solutionbox/code_free_ml/mltoolbox/code_free_ml/analyze.py]: not if the pattern is on GCS.

```
In [73]: %%time
    ...: file_io.get_matching_files('gs://cloud-ml-dev_bdt/*')
    ...: 
CPU times: user 1.36 s, sys: 128 ms, total: 1.49 s
Wall time: 12.2 s
Out[73]: 
['gs://cloud-ml-dev_bdt/analyze_out', ..., ]
```

Comment (brandondutra) [inline solutionbox/code_free_ml/mltoolbox/code_free_ml/analyze.py]: ok, well, 12 secs is 'quick', but in terms of thinking printing 'analyze file' has lag, it's useful to show there is some startup costs.

Comment (qimingj) [inline solutionbox/code_free_ml/mltoolbox/code_free_ml/analyze.py]: oh.. Makes sense.

**Pull request merged**